cargo/cargo_build_script_runner/lib.rs

```diff
@@ -0,0 +1,159 @@
+        let split = line.splitn(2, '=').collect::<Vec<_>>();
+        if split.len() <= 1 {
+            return BuildScriptOutput::None;
+        }
+        let param = split[1].trim().to_owned();
+        match split[0] {
+            "cargo:rustc-link-lib" => BuildScriptOutput::LinkLib(param),
+            "cargo:rustc-link-search" => BuildScriptOutput::LinkSearch(param),
+            "cargo:rustc-cfg" => BuildScriptOutput::Cfg(param),
+            "cargo:rustc-flags" => BuildScriptOutput::Flags(param),
+            "cargo:rustc-env" => BuildScriptOutput::Env(param),
+            _ => BuildScriptOutput::None,
+        }
+    }
+
+    /// Converts a [BufReader] into a vector of [BuildScriptOutput] enums.
+    fn from_reader<T: Read>(mut reader: BufReader<T>) -> Vec<BuildScriptOutput> {
+        let mut result = Vec::<BuildScriptOutput>::new();
+        let mut line = String::new();
+        loop {
+            line.clear();
+            if reader.read_line(&mut line).expect("Valid script output") == 0 {
+                return result;
+            }
+            let bso = BuildScriptOutput::new(&line);
+            if bso != BuildScriptOutput::None {
+                result.push(bso);
+            }
+        }
+    }
+
+    /// Take a [Command], execute it and converts its input into a vector of [BuildScriptOutput]
+    pub fn from_command(cmd: &mut Command) -> Vec<BuildScriptOutput> {
+        let mut child = cmd.stdout(Stdio::piped()).spawn().expect("Unable to start binary");
+        let ecode = child.wait().expect("failed to wait on child");
+        let reader = BufReader::new(
+                child
+                .stdout
+                .as_mut()
+                .expect("Failed to open stdout"),
+            );
+        assert!(ecode.success());
+        Self::from_reader(reader)
+    }
+
+    /// Convert a vector of [BuildScriptOutput] into a list of environment variables.
+    pub fn to_env(v: &Vec<BuildScriptOutput>) -> String {
+        v.iter()
+            .filter_map(|x| {
+                if let BuildScriptOutput::Env(env) = x {
+                    Some(env.to_owned())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>()
+            .join(" ")
+    }
+
+    /// Convert a vector of [BuildScriptOutput] into a flagfile.
+    pub fn to_flags(v: &Vec<BuildScriptOutput>) -> String {
+        v.iter()
+            .filter_map(|x| match x {
+                BuildScriptOutput::Cfg(e) => Some(format!("--cfg={}", e)),
+                BuildScriptOutput::Flags(e) => Some(e.to_owned()),
+                BuildScriptOutput::LinkLib(e) => Some(format!("-l{}", e)),
+                BuildScriptOutput::LinkSearch(e) => Some(format!("-L{}", e)),
+                _ => None,
+            })
+            .collect::<Vec<_>>()
+            .join(" ")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Cursor;
+
+    #[test]
```

Comment:
WDYT about adding a test for invalid build script output?

Reply:
Done.